quickfixj-core/src/main/java/quickfix/Message.java

```diff
@@ -581,9 +581,11 @@ private void parseGroup(String msgType, StringField field, DataDictionary dd, Da
 		// QFJ-533
         int declaredGroupCount = 0;
         try {
-          declaredGroupCount = Integer.parseInt(field.getValue());
+             declaredGroupCount = Integer.parseInt(field.getValue());
         } catch (final NumberFormatException e) {
-            throw new InvalidMessage("NumberFormatException: Repeating group count requires an Integer but found:" +field.getValue());
+			 InvalidMessage invalidMessage = new InvalidMessage("Repeating group count requires an Integer but found:" +field.getValue());
+			 invalidMessage.initCause(e);
+		     throw invalidMessage;
         }
```

Comment:
Does InvalidMessage not have a constructor `InvalidMessage(String message, Throwable cause)`

Reply:
No I can just find the constructor:  InvalidMessage(String message)